src/Ninject.Web.AspNetCore/NinjectServiceProviderBuilder.cs

```diff
@@ -16,7 +17,26 @@ public NinjectServiceProviderBuilder(AspNetCoreKernel kernel, IServiceCollection
 
 		public IServiceProvider Build()
 		{
-			_kernel.Bind<IServiceProvider>().ToConstant(_kernel.RootScope.ServiceProvider);
+			_kernel.Bind<IServiceProvider>().ToMethod(context => {
+				// Another interesting quirk of the Microsoft DI framework is that injecting the IServiceProvider injects a different
+				// instance depending on the context. EXCEPT if the service provider is injected into a singleton in which case the
+				// root service provider is used...
+				//
+				// It essentially always injects the instance that is being used for the instantiation
+				// which kinda makes sense, but is not documented ANYWHERE, not tested ANYWHERE and was really quite difficult to
+				// figure out.
+				var scopeProvider = context.Parameters.OfType<ServiceProviderScopeParameter>().SingleOrDefault()?.SourceServiceProvider;
```

Comment:
This is the core fix for issue No. 1 together with the changes to the ServiceProviderScopeParameter which acually allows us to get the service provider of the scope.

Reply:
Great, that a quite localized change helped to solve the issue. Very nice.